parla/tasks.py

```diff
@@ -447,6 +447,124 @@ def get_current_devices() -> List[Device]:
     return list(task_runtime.get_devices())
 
 
+@contextmanager
+def _reserve_persistent(memsize, device):
+    resource_pool = get_scheduler_context().scheduler._available_resources
+    resource_pool.allocate_resources(device, {'memory' : memsize}, blocking = True)
+    try:
+        yield
+    finally:
+        resource_pool.deallocate_resources(device, {'memory' : memsize})
+
+
+def _get_parla_device(device):
+    if isinstance(device, Device):
+        return device
+    try:
+        import cupy
+    except ImportError:
+        pass
+    else:
+        if isinstance(device, cupy.cuda.Device):
+            from .cuda import gpu
+            index = device.id
+            return gpu(index)
+    raise ValueError("Don't know how to convert object of type {} to a parla device object.".format(type(device)))
+
+
+@contextmanager
+def reserve_persistent_memory(amount, device = None):
```

Comment:
OK for now, but add TODO. This should not be specific to memory, it should be possible to reserve any resource, so this should be called `reserve` or `reserve_resources` and should take keyword arguments for all resources (maybe via `**`).

Reply:
Good point.